### PR TITLE
Next batch of editorial issues

### PR DESCRIPTION
**Pull request by ptomato**
Best reviewed commit by commit. Explanations in each commit message. Closes several issues opened by @anba.

### CODE REVIEW

**Review by gibson042 — APPROVED** [review]
LGTM, thanks!

File: polyfill/lib/ecmascript.mjs [hunk 1]
```diff
@@ -1452,8 +1452,7 @@ export function ToTemporalMonthDay(item, options) {
     if (calendar !== 'iso8601') {
       throw new Error(`assertion failed: missing year with non-"iso8601" calendar identifier ${calendar}`);
     }
-    RejectISODate(1972, month, day);
-    return CreateTemporalMonthDay(month, day, calendar);
+    return CreateTemporalMonthDay(month, day, calendar, 1972);
```
Comment (justingrant) [inline polyfill/lib/ecmascript.mjs:1455]: Should 1972 be a named constant instead of a number literal?  Might be clearer that way.

**Review by justingrant — APPROVED** [review]
Looks great, thanks!

**Review by jasonwilliams — APPROVED** [review]
LGTM

**Pull request merged**